### ISSUE TITLE
Merging types breaks refs on conflicts

### ISSUE THREAD

**Issue by fuzyll** [Vector35/binaryninja-api#5722]
If I have:

```
struct bbbb {        
    int aaaa;        
}                    
struct aaaa {        
    struct bbbb aaaa;
}                    
```

...defined newly on both sides of a merge, it should merge into two structs with suffixes and updated references:

```
struct bbbb_1 {            struct bbbb_2 {
    int aaaa;                  int aaaa;
}                          }
struct aaaa_1 {            struct aaaa_2 {
    struct bbbb_1 aaaa;        struct bbbb_2 aaaa;
}                          }
```

It does the two structs with suffixes part correctly, but leaves the inner member still pointing to `bbbb` which no longer exists.

Blocked by #5720.
